src/openforms/formio/api/validators.py

```diff
@@ -111,6 +120,11 @@ def __call__(self, value: UploadedFile) -> None:
             "image/heif",
         ):
             return
+        # 4795
+        # The sdk cannot determine the file type of .msg files, which result into
+        # content_type "". So we have to validate these for ourselves
+        elif mime_type == "application/vnd.ms-outlook" and ext == "msg":
+            return
```

Comment:
rather than adding another hardcoded check here, we can do better, but we need to restructure the validator code a bit!

First, start with figuring out the provided content type:

```py
provided_content_type = value.content_type or "application/octet-stream"
```

and replace all `value.content_type` checks with `provided_content_type`.

The `or` clause is important here - in situations where the mime type could not be determined by the browser and it returns an empty string, the backend will fall back to the generic binary data mime type. But, the interesting part is that in the generic binary data check:

```py
if value.content_type == "application/octet-stream": ...
```
the extensions for the binary data are looked up and checked against the provided file extension. So, we invert the process here - given the binary data, what should the extension be, and validate this.